src/hooks.ts

```diff
@@ -75,13 +75,22 @@ export function useProgress(updateInterval?: number) {
   const [state, setState] = useState<ProgressState>({ position: 0, duration: 0, buffered: 0 })
   const playerState = usePlaybackState()
   const stateRef = useRef(state)
+  const isUnmountedRef = useRef(true)
+  useEffect(() => {
+    isUnmountedRef.current = false
+    return () => {
+      isUnmountedRef.current = true
+    }
+  }, [])
 
   const getProgress = async () => {
     const [position, duration, buffered] = await Promise.all([
       TrackPlayer.getPosition(),
       TrackPlayer.getDuration(),
       TrackPlayer.getBufferedPosition(),
     ])
+    // After the asynchronous code is executed, if the component has been uninstalled, do not update the status
+    if (isUnmountedRef.current) return
```

Comment:
is there a need for this to happen after the async call? Did you find that the un-mounting happens while the async call is still executing?

Reply:
Yes, this problem may occur when the status is updated after all asynchronous calls.
Although it may be rare, see: <https://www.debuggr.io/react-update-unmounted-component/>